backend/routers/auth.py

```
@@ -0,0 +1,81 @@
+    if session:
+        _set_session_cookies(response, session)
+    user_payload = user.model_dump() if user and hasattr(user, "model_dump") else None
+    return {"user": user_payload, "message": "Signup complete"}
+
+@router.post("/auth/logout")
+async def auth_logout(response: Response):
+    _clear_session_cookies(response)
+    return {"ok": True}
+
+@router.get("/auth/me")
+async def auth_me(user: dict = Depends(get_current_user)):
```

Ruff (0.15.15)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)